internal/tfengine/tfengine.go

```diff
@@ -272,3 +279,55 @@ func ConvertToLocalBackend(path string) error {
 
 	return nil
 }
+
+// findUnmanaged walks both directories and shows all files in the output
+// directory which aren't in the generated files directory.
+// If the deleteFiles flag is set, it will also delete these files.
+func findUnmanaged(generatedDir, outputDir string, deleteFiles bool) error {
+	managed := make(map[string]bool)
+
+	mkFn := func(addToManaged bool, root string) func(path string, info os.FileInfo, err error) error {
+		root = filepath.Clean(root)
+		return func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return fmt.Errorf("walk path %q: %v", path, err)
+			}
+
+			trimmedPath := strings.TrimPrefix(filepath.Clean(path), root)
+			if addToManaged {
+				managed[trimmedPath] = true
+				return nil
+			}
+
+			if _, ok := managed[trimmedPath]; ok {
+				return nil
+			}
+
+			if !deleteFiles {
+				log.Printf("WARNING: Unmanaged file: %v\n", path)
+				return nil
+			}
+
+			fullPath, err := fileutil.Expand(path)
+			if err != nil {
+				return err
+			}
+			log.Printf("WARNING: Deleting unmanaged file: %v\n", path)
```

Comment:
In a situation where the files are generated successfully once or were present in the output directory before, and then another tfengine run failed, with the current logic, it would list all the previously generated files from the output directory as unmanaged files (although they were actually managed files) iiuc. Is there a way to prevent that or avoid that confusion?

Reply:
From what I can tell from the code, it won't get to this step if it failed to generate the files, it would return an error early. The only time it would get errors and still look for unmanaged files is if adding licenses or formatting fails, but none of those will affect the file names, so it shouldn't print all files as if they were unmanaged.

Are you able to produce a case where it lists all previously-generated files?

Reply:
You are right. Shouldn't get to this step if failed to generated files.